main/src/library/File.flix

```diff
@@ -119,6 +119,55 @@ namespace File {
                 Err(getMessage(ex))
         }
 
+    ///
+    /// Returns a string containing the given file `f`.
+    ///
+    pub def read(f: String): Result[String, String] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:readString(##java.nio.file.Path);
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let result = readString(javaPath);
+            Ok(result)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns a string containing the given file `f` with the options `opts`.
+    /// The options `opts` to apply consists of
+    /// `offset` - the start offset in the given file `f`.
+    /// `count` - the number of bytes read.
+    /// `charSet` - the specific charset to be used to decode the bytes.
+    ///
+    pub def readWith(opts: {offset :: Int64, count :: Int32, charSet :: String}, f : String): Result[String, String] & Impure = {
+        try {
+            import new java.io.File(String) as newFile;
+            import new java.io.FileInputStream(##java.io.File) as newInputStream;
+            import java.io.FileInputStream.skip(Int64);
+            import java.io.FileInputStream.read(Array[Int8], Int32, Int32);
+            import java.nio.charset.Charset:forName(String);
+            import new java.lang.String(Array[Int8], ##java.nio.charset.Charset) as newString;
+
+            let javaFile = newFile(f);
+            let stream = newInputStream(javaFile);
+            skip(stream, opts.offset);
+            let bytes = [0i8; opts.count];
+            read(stream, bytes, 0, opts.count);
```

Comment:
Should I do some error handling here? If read throws an exception

Reply:
While looking up the exceptions I re-read the documentation, it says "Reads up to len bytes of data from this input stream into an array of bytes. If len is not zero, the method blocks until some input is available; otherwise, no bytes are read and 0 is returned."

I don't think read can actually solve this problem since the offset only relates to where in the byte array it should put the data. Maybe the easiest fix is to use `FileInputStream.skip(offset)` (possible `offset-1` maybe?) and then just call the read function with `offset = 0`. I'm not sure

Reply:
I realized the same when testing the readWith, and I found the same solution as you are suggesting :)